elements/sdk/eos-knowledge-lib.bst

```diff
@@ -0,0 +1,38 @@
+kind: autotools
+
+build-depends:
+- freedesktop-sdk.bst:public-stacks/buildsystem-autotools.bst
+- sdk-build-depends/eos-hotdoc-theme.bst
+- sdk-build-depends/hotdoc-modular-framework.bst
+- sdk-build-depends/hotdoc.bst
+- sdk-build-depends/jasmine-gjs.bst
+
+depends:
+- freedesktop-sdk.bst:components/gstreamer.bst
+- gnome-sdk.bst:sdk/WebKitGTK.bst
+- gnome-sdk.bst:core/evince.bst
+- gnome-sdk.bst:sdk/gjs.bst
+- sdk-depends/emeus.bst
+- sdk-depends/eos-knowledge-content-renderer.bst
+- sdk-depends/mathjax.bst
+- sdk-depends/maxwell.bst
+- sdk-depends/python-polib.bst
+- sdk-depends/python-pyyaml.bst
+- sdk-depends/sassc.bst
+- sdk/eos-sdk.bst
+- sdk/libdmodel.bst
+
+variables:
+  conf-local: |
+    --disable-js-doc \
+    --disable-documentation \
+    --with-mathjax-dir="%{datadir}/javascript/mathjax"
+
+environment:
+  HOTDOC: "%{bindir}/python3 %{bindir}/hotdoc"
+
+sources:
+- kind: git_tag
+  url: github_com:endlessm/eos-knowledge-lib.git
+  track: sdk-6
+  ref: Release_3.2.6-699-g7de89777bf721bd257e2b9240401d18f8a8add12
```

Comment:
Since we don't actually tag our SDK dependencies, does it make sense to specify this? Clearly this is just random `git-describe` output and not an actual tag. https://gitlab.gnome.org/GNOME/gnome-build-meta/blob/master/elements/sdk/glib.bst for instance doesn't specify `ref`, presumably because they want to just use whatever is the latest on `master`.

I actually don't really understand why the `git_tag` element is preferred over the regular `git` element. Both seem to handle git tags. @bochecha?

Reply:
The idea is a particular commit in endless-sdk-flatpak is tracking a particular commit in each repo, and that ref changes (according to the value of the _track_ property) whenever we run `bst track`. It is possible to track at build time (using a separate project.refs file which is gitignored). That's what gnome-build-meta uses for their _master_ branch. For release branches they use either the same format as us, or use download URLs instead of git commits.

Good to ask about git vs git_tag :) As I understand it git_tag adds the "track" property, which is helpful to control what gets tracked automtically - otherwise those refs get kind of fiddly.

Reply:
On tracking at build time, I predict that we may want to discuss that at some point, but for our little SDK I imagine we'll be fine without as long as we have some replacement for the per-component "flapjack-" jobs.